lib/slime/preprocessor.ex

```diff
@@ -25,6 +26,35 @@ defmodule Slime.Preprocessor do
     String.replace(document, ~r/\n+\z/m, "")
   end
 
+  @collapse_backslash_statements_regex ~r/^([-=])(.*)\\$\s+/m
+  @collapse_comma_statements_regex ~r/^([-=])(.*),$\s+/m
+
+  defp collapse_statements(document) do
+    collapse_backslash_statements collapse_comma_statements(document)
+  end
+
+  defp collapse_backslash_statements(document) do
```

Comment:
I think it might be possible to refactor these two, `collapse_backslash_statements/1` and `collapse_comma_statements/1`, into one.  What do you think about something like this:

```elixir
defp collapse_statements(document, regex, replacement) do
  cond do
    String.match?(document, regex) ->
      document
      |> String.replace(regex, replacement)
      |> collapse_statements(regex, replacement)
    true ->
      document
  end
end
```

It would require a slight change to `collapse_statements/1`:

```elixir
defp collapse_statements(document) do
  document
  |> collapse_statements(@collapse_comma_statements_regex, "\\1\\2,")
  |> collapse_statements(@collapse_backslash_statements_regex, "\\1\\2")
end
```